components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/step/impl/DefaultStepHandler.java

```diff
@@ -72,7 +74,12 @@ public void handle(HttpServletRequest request, HttpServletResponse response,
 
         List<AuthenticatorConfig> authConfigList = stepConfig.getAuthenticatorList();
 
-        String authenticatorNames = FrameworkUtils.getAuthenticatorIdPMappingString(authConfigList);
+        String authenticatorNames;
+        try {
+            authenticatorNames = URLEncoder.encode(FrameworkUtils.getAuthenticatorIdPMappingString(authConfigList), "UTF-8");
```

Comment:
Here I have not removed encoding from all the places in login.jsp. Instead I encoded 'idp name' in a central place and used it without repeating. According to the previous comment removed the central encoding and encode it at the places we render the output.